source/code/plugins/oms_common.rb

```diff
@@ -478,6 +482,118 @@ class Common
+        return false unless qwa.length == 4
+        return false unless qwa[0].to_i != 0
+        i = 0
+        qwa.each do |quadwordstring|
+            bi = quadwordstring.to_i
+            # This may need more detail if 255 octets are sometimes allowed, but I don't think so.
+            return false unless 0 <= bi and bi < 255
+            i += 1
+        end
+        return true
+      end
+
+      def is_like_ipv6_string?(hnBuffer)
+        return true if /\A#{IPV6_REGEX}\z/ =~ hnBuffer
+        return false
+      end
+
+      def look_for_socket_class_host_address
+        hostname_buffer = nil
+
+        begin
+          hostname_buffer = Socket.gethostname
+        rescue => error
+          OMS::Log.error_once("Unable to get the Host Name using socket facility: #{error}")
+          return
+        end
+        hostname_buffer = clean_hostname_string(hostname_buffer)
+
+        if is_like_ipv4_string?(hostname_buffer) then
+          # If Socket.gethostname returns an IPv4 address,
+          # use the entire string.
+          @@Hostname = hostname_buffer
+        else
+          # Here we presume we got a usable hostname,
+          # though it could also be an IPV6, or other things,
+          # and more attention may be needed here later.
+          @@Hostname = clean_hostname_string(hostname_buffer)
+        end
+        return # Thwart accidental return to force correct use.
+      end
+
+      def look_in_designated_hostnamefile
+        # Issue:
+        #   When omsagent runs inside a container, gethostname returns the hostname of the container (random name)
+        #   not the actual machine hostname.
+        #   One way to solve this problem is to set the container hostname same as machine name, but this is not
+        #   possible when host-machine is a private VM inside a cluster.
+        # Solution:
+        #   Share/mount ‘/etc/hostname’ as '/var/opt/microsoft/omsagent/state/containername' with container and
+        #   omsagent will read hostname from shared file.
+
+        unless File.readable?(@@HostnameFilePath)
+            OMS::Log.warn_once("File '#{@@HostnameFilePath}' exists but is not readable.")
+            return
+        end
+
+        begin
+          hostname_buffer = File.read(@@HostnameFilePath)
+        rescue => error
+          OMS::Log.warn_once("Unable to read the hostname from #{@@HostnameFilePath}: #{error}")
+        end
+        @@Hostname = clean_hostname_string(hostname_buffer)
+        return # Thwart accidental return to force correct use.
+      end
+
+      def validate_hostname_equivalent(hnBuffer)
+        # RFC 2181
+        rfcl = "RFC 2181 specified hostname range of {1,63} octets for non-root item."
+        return if is_hostname_compliant?(hnBuffer)
+        return if is_like_ipv4_string?(hnBuffer) 
+        return if is_like_ipv6_string?(hnBuffer)
+        msg = "Hostname '#{hnBuffer}' not compliant (#{rfcl}).  Not IP Address Either."
+        OMS::Log.error_once(msg)
```

Comment:
I believe this should just be a warning - we will still handle this data, but the customer should be aware something is unexpected.

Reply:
Last log is a warning on this.  I'll change this for now, but it seemed like an error that we just handled, and that might be well labeled with higher seriousness at the exception point.  I'll set this back to warn for now.